### PR TITLE
feat: add blog og image

### PR DESCRIPTION
**Pull request by dephraiim**
Preview of OG Image

CleanShot 2023-08-30 at 02 47 34@2x

CleanShot 2023-08-30 at 02 52 56@2x

### CODE REVIEW

**Comment by Mythie** [comment]
Love it 💚

File: apps/marketing/src/app/(marketing)/blog/[post]/opengraph-image.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    ),
+    fetch(new URL('./../../../../assets/inter-regular.ttf', import.meta.url)).then(async (res) =>
+      res.arrayBuffer(),
+    ),
+    fetch(new URL('./../../../../assets/background-blog-og.png', import.meta.url)).then(
+      async (res) => res.arrayBuffer(),
+    ),
+    fetch(new URL('./../../../../../public/logo.png', import.meta.url)).then(async (res) =>
+      res.arrayBuffer(),
+    ),
```
Comment (Mythie) [inline apps/marketing/src/app/(marketing)/blog/[post]/opengraph-image.tsx:37]: These still make me sad but there isn't much we can do I think

Comment (adithyaakrishna) [reply apps/marketing/src/app/(marketing)/blog/[post]/opengraph-image.tsx:37]: How about if we kinda make it like,

```typescript
const assetUrls = [
  './../../../../assets/inter-bold.ttf',
  './../../../../assets/inter-regular.ttf',
  './../../../../assets/background-blog-og.png',
  './../../../../../public/logo.png',
];

async function fetchArrayBuffers(urls) {
  const responses = await Promise.all(urls.map(url => fetchArrayBuffer(url)));
  return responses;
}

const resourceArrayBuffers = await fetchArrayBuffers(assetUrls);
const [interBold, interRegular, backgroundImage, logoImage] = resourceArrayBuffers;
```

Comment (Mythie) [reply apps/marketing/src/app/(marketing)/blog/[post]/opengraph-image.tsx:37]: That's still just fluff around what's happening, my sadness is mostly around needing a ton of `../../../` to get it working

Comment (adithyaakrishna) [reply apps/marketing/src/app/(marketing)/blog/[post]/opengraph-image.tsx:37]: Yeah, I agree, though this make it less scary, the core issue still remains

Comment (Mythie) [reply apps/marketing/src/app/(marketing)/blog/[post]/opengraph-image.tsx:39]: Did the tidy, broke generation of images due to the compiler intercepting these `import.meta.url` calls. Will leave it for now :smile:

**Review by Mythie — APPROVED** [review]

**Pull request merged**